### ISSUE TITLE
[base] 13: Ability to manage ondelete attribute for m2m from odoo for custom fields

### ISSUE THREAD

**Issue by sswapnesh** [odoo/odoo#49736]
Impacted versions: 13.0

Steps to reproduce: Create new m2m field from Settings --> Fields

Current behavior: No option to manage ondelete attribute 
![screenshot-localhost_8069-2020 04 17-23_19_30](https://user-images.githubusercontent.com/16624719/79598668-fc88de00-8101-11ea-9c95-b7c641d7a7df.png)

Expected behavior: Possible to manage ondelete attributes for custom field

Video/Screenshot link (optional):

Ps: ondelete on m2m is supported since https://2241006-13-0-143960.runbot26.odoo.com/web

**Comment by pedrobaeza** [comment]
This is totally wishlist for me, and it's not a good maintenance practice to add custom fields through UI.

**Comment by sswapnesh** [comment]
Yes a Wishlist,  Well sometimes we need it as Lots of users  (Specially Enterprise ones) are creating fields from UI.

**Comment by pedrobaeza** [comment]
As said, IMO you should forbid that or not give them support if you don't want a hell.

**Comment by Elkasitu** [comment]
Not sure about the reason in this specific case but it may have been done on purpose, as an example Selection fields don't support ondelete attribute on UI either because it creates a ton of complexity for no added value

**Comment by sswapnesh** [comment]
@Elkasitu Added #50977

Can you review it?


At least I'm sure we can merge this part 

`'on_delete': field.ondelete if field.type in ['many2one', 'many2many'] else None,` and fixing domain on `on_delete`



cc @seb-odoo based on 064ebe3b1aa359655f7e4966e682c15cc711e9f9
